components/prism-pascal.js

```diff
@@ -5,9 +5,17 @@
 */
 
 Prism.languages.pascal = {
+	'asm': {
+		pattern: /(asm)([\s\S]+?)(end)/i,
```

Comment:
One of the tests fails because of the unnecessary capturing groups, so ket's fix that and let's add word boundary assertions as well.
```suggestion
		pattern: /(asm\b)[\s\S]+?\bend/i,
```

Reply:
Thanks for your reply and suggestions